[RDY] Refactor vip

Alberth289346 opened the pull request.

TheCycoONE commented on CorsixTH/Lua/entities/vip.lua:
```diff
@@ -255,6 +255,22 @@ function Vip:evaluateEmergency(success)
+    end
+  end
+  -- Should never get here, but just in case
+  return 0
```
Should assert or print paranoia debug output here, don't want exception cases to be silent.

TheCycoONE commented on CorsixTH/Lua/entities/vip.lua:
```diff
@@ -255,6 +255,22 @@ function Vip:evaluateEmergency(success)
+--! upper bound (leave out for the final entry), and the value to return if
+--! the provided value is at or below the upper bound.
+--!return Evaluation value.
+local function evaluateProperty(value, eval_table)
```
name isn't great. Maybe rangeMapLookup?
Could possibly be general enough (or generalized enough) to move up to utilities.

TheCycoONE commented on CorsixTH/Lua/entities/vip.lua:
```diff
@@ -265,15 +281,14 @@ function Vip:setVIPRating()
-    elseif deathRatio < 2 then
-      self.vip_rating = self.vip_rating - 20
-    end
+    local death_ratio_table = {
```
If we agree to call this a range map above I'd like to keep that name here - e.g. death_ratio_rangemap.

TheCycoONE commented:
:+1: from sight inspection of all the averages changes.

Alberth289346 commented:
Generalized the rangemap function, and moved it to utils; merged all the averaging changes into the 2nd commit. Aside from 3 additional comments "default value does not generate a message", it's equivalent.
Added new commit to rename camel case variables to lower_underscore_case to unify the variable names somewhat. Tested with a level 1 save, with a VIP visit, giving a report.
